pkg/cmd/render/render.go

```diff
@@ -377,6 +382,42 @@ func (t *TemplateData) setPlatform(infraConfigFilePath string) error {
 	return nil
 }
 
+func (t *TemplateData) setUnsupportedKeyForCR(etcdOperatorCR string) error {
+	type UnSupported struct {
+		UseUnsupportedUnsafeNonHANonProductionUnstableEtcd bool `json:"useUnsupportedUnsafeNonHANonProductionUnstableEtcd"`
+	}
+	etcdOperatorCRStructure := &operatorv1.MyOperatorResource{}
+	etcdOperatorCRData, err := ioutil.ReadFile(etcdOperatorCR)
+	if err != nil {
+		return err
+	}
+	etcdJson, err := yaml.YAMLToJSON(etcdOperatorCRData)
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(etcdJson, etcdOperatorCRStructure)
```

Comment:
You can probably use the `unstructured`[1] library to make this simpler/closer to the pattern seen around.

```
etcd := &unstructured.Unstructured{}
err := etcd.UnmarshalJSON([]byte(etcdJson))
...
err := unstructured.SetNestedField(etcd, true, "spec", "unsupportedConfigOverrides", "UseUnsupportedUnsafeNonHANonProductionUnstableEtcd")
...
// get json with etcd.MarshalJSON 
```
1. https://github.com/kubernetes/apimachinery/blob/15d95c0b2af3f4fcf46dce24105e5fbb9379af5a/pkg/apis/meta/v1/unstructured/helpers.go#L208:6